R/mergedatasetsbycase.R

```diff
@@ -2580,14 +2582,30 @@ variableLabelFromDataSets <- function(matched.names.row, data.sets,
     return("")
 }
 
-# Create `Source of cases` variable for the output file, which allows the consumer to determine which input file each variable came from.
-mergeSrc <- function(n.data.set.cases, data.set.names)
+# Create `Source of cases` variable for the output file,
+# which allows the consumer to determine which input file
+# each variable came from.
+mergeSrc <- function(n.data.set.cases, data.set.names, existing.mergesrc = NULL)
 {
+    if (!is.null(existing.mergesrc)) {
+        em.labels <- names(attr(existing.mergesrc, "labels"))
+        existing.mergesrc <- em.labels[existing.mergesrc]
+    }
+
     n.data.sets <- length(n.data.set.cases)
     result <- rep(seq_len(n.data.sets), n.data.set.cases)
+    file.names <- data.set.names
+    result <- file.names[result]
```

Comment:
It would be good to use a different variable with a descriptive name instead of reassigning `result` a few times.